daffodil-core/src/test/scala/org/apache/daffodil/dsom/TestRefMap.scala

```diff
@@ -753,29 +753,51 @@ class TestRefMap {
             <xs:element name="f" type="xs:int" dfdl:inputValueCalc='{ ../../rr/e }' />
           </xs:sequence>
         </xs:group>
+        <xs:group name="x">
+          <xs:sequence>
+            <!--
+            likewise, this group is never used, and so should not create an error, even
+            if checkAllTopLevel is true
+            -->
+            <xs:element name="y" type="xs:int" dfdl:outputValueCalc='{ ../../dne }'/>
+            <xs:element name="z" type="xs:int" dfdl:inputValueCalc='{ ../../dne }' />
+          </xs:sequence>
+        </xs:group>
 
   /**
    * The root element doesn't have any substructure that eventually uses the
-   * group h, so we get the error that indicates h can't be a root (relative path upward)
-   * because we're checking all top level.
+   * group h, and so the IVC/OVC expressions within it which reach passed the
+   * root are not validated. This is because such expressions potentially only
+   * make sense within the context of the root element. checkAllTopLevel does
+   * not mean that all elements that aren't used must be make sense as root
+   * elements--even though "h" could be a root, because it isn't referenced
+   * anywhere, that doesn't necessarily mean we should require that expressions
+   * inside it as if it were a root.
```

Comment:
phrasing of last line. I think this needs to say "that expressions inside it are not up-and-out expressions, as if it were a root."